server/server_test.go

```diff
@@ -603,32 +603,46 @@ func TestNilMonitoringPort(t *testing.T) {
 	}
 }
 
-type DummyAuth struct{}
+type DummyAuth struct {
+	t         *testing.T
+	needNonce bool
+}
 
 func (d *DummyAuth) Check(c ClientAuthentication) bool {
+	if d.needNonce && len(c.GetNonce()) == 0 {
```

Comment:
It's ok since tests are not supposed to fail, but if it fails, it will produce a DATA race since t.Fatal will be invoked from different go routine than the test itself. Again, no worry since if we get the DATA race it means that the test failed anyway.

Reply:
yeah fair, but its exceptional as you say so if its fails its failing anyway :) I'll leave as is and merge on green